File: java/src/Creational/Builder/SoupBuffet.java
package Creational.Builder;


//SoupBuffet.java - A helper class

class SoupBuffet {
   String soupBuffetName;
   
   ChickenSoup chickenSoup;
   ClamChowder clamChowder;
   FishChowder fishChowder;
   Minnestrone minnestrone;
   PastaFazul pastaFazul;
   TofuSoup tofuSoup;
   VegetableSoup vegetableSoup;

   public String getSoupBuffetName() {
       return soupBuffetName;
   }
   public void setSoupBuffetName(String soupBuffetNameIn) {
      soupBuffetName = soupBuffetNameIn;
   }
   
   public void setChickenSoup(ChickenSoup chickenSoupIn) {
       chickenSoup = chickenSoupIn;
   }
   public void setClamChowder(ClamChowder clamChowderIn) {
       clamChowder = clamChowderIn;
   }
   public void setFishChowder(FishChowder fishChowderIn) {
      fishChowder = fishChowderIn;
   }
   public void setMinnestrone(Minnestrone minnestroneIn) {
      minnestrone = minnestroneIn;
   }
   public void setPastaFazul(PastaFazul pastaFazulIn) {
       pastaFazul = pastaFazulIn;
   }
   public void setTofuSoup(TofuSoup tofuSoupIn) {
       tofuSoup = tofuSoupIn;
   }
   public void setVegetableSoup(VegetableSoup vegetableSoupIn) {
       vegetableSoup = vegetableSoupIn;
   }
   
   public String getTodaysSoups() {
        StringBuffer stringOfSoups = new StringBuffer();
        stringOfSoups.append(" Today's Soups!  ");
        stringOfSoups.append(" Chicken Soup: ");        
        stringOfSoups.append(this.chickenSoup.getSoupName()); 
        stringOfSoups.append(" Clam Chowder: ");        
        stringOfSoups.append(this.clamChowder.getSoupName()); 
        stringOfSoups.append(" Fish Chowder: ");        
        stringOfSoups.append(this.fishChowder.getSoupName()); 
        stringOfSoups.append(" Minnestrone: ");        
        stringOfSoups.append(this.minnestrone.getSoupName());
        stringOfSoups.append(" Pasta Fazul: ");        
        stringOfSoups.append(this.pastaFazul.getSoupName());
        stringOfSoups.append(" Tofu Soup: ");        
        stringOfSoups.append(this.tofuSoup.getSoupName());
        stringOfSoups.append(" Vegetable Soup: ");        
        stringOfSoups.append(this.vegetableSoup.getSoupName());
        return stringOfSoups.toString();          
   }
}        

File: go/creational/common/SoupBuffet.go

package common

// used by creational builder, and creational factorymethod

import (
	"fmt"
)

type SoupBuffet struct {
   soupBuffetName string
   ChickenSoup Soup
   ClamChowder Soup
   FishChowder Soup
   Minnestrone Soup
   PastaFazul Soup
   TofuSoup Soup
   VegetableSoup Soup
}

func NewSoupBuffet() *SoupBuffet {
	return &SoupBuffet{}
}

func (this *SoupBuffet) GetSoupBuffetName() string {
	return this.soupBuffetName
}

func (this *SoupBuffet) SetSoupBuffetName(soupBuffetNameIn string) {
	this.soupBuffetName = soupBuffetNameIn
}
   
func (this *SoupBuffet) SetChickenSoup(chickenSoupIn Soup) {
	 this.ChickenSoup = chickenSoupIn
}

func (this *SoupBuffet) SetClamChowder(clamChowderIn Soup) {
	 this.ClamChowder = clamChowderIn
}

func (this *SoupBuffet) SetFishChowder(fishChowderIn Soup) {
	this.FishChowder = fishChowderIn
}

func (this *SoupBuffet) SetMinnestrone(minnestroneIn Soup) {
	this.Minnestrone = minnestroneIn
}

func (this *SoupBuffet) SetPastaFazul(pastaFazulIn Soup) {
	 this.PastaFazul = pastaFazulIn
}

func (this *SoupBuffet) SetTofuSoup(tofuSoupIn Soup) {
	 this.TofuSoup = tofuSoupIn
}

func (this *SoupBuffet) SetVegetableSoup(vegetableSoupIn Soup) {
	 this.VegetableSoup = vegetableSoupIn
}
   
func (this *SoupBuffet) GetTodaysSoups() string {
	return fmt.Sprintf(" Today's Soups!   Chicken Soup: %s Clam Chowder: %s Fish Chowder: %s Minnestrone: %s Pasta Fazul: %s Tofu Soup: %s Vegetable Soup: %s" ,
		  this.ChickenSoup.GetSoupName() ,
        this.ClamChowder.GetSoupName() ,
        this.FishChowder.GetSoupName() ,
        this.Minnestrone.GetSoupName(),
        this.PastaFazul.GetSoupName(),
        this.TofuSoup.GetSoupName() ,
        this.VegetableSoup.GetSoupName() )
}        
